packages/snap/src/generate-openapi.ts

```diff
@@ -0,0 +1,175 @@
+
+import { LockedData } from '@motiadev/core';
+import * as fs from 'fs';
+import { OpenAPIV3 } from 'openapi-types';
+import * as path from 'path';
+import { generateLockedData } from './generate-locked-data';
+
+export async function generateOpenApi(projectDir: string, title?: string, version?: string) {
```

Comment:
let's move these functions to a separate folder, it will be easier to maintain later, and to write tests, we need tests for this, make sure to add test cases for some edge cases like anyof (with non nullable types) and nulllable

```
openapi/
  generate-openapi.ts
  process-schema.ts
  openapi-utils.ts
```

for tests, create a folder `__tests__/` under `src/` and add `openapi/` inside

```
src/
  __tests__/
    openapi/
      generate-openapi.spec.ts
      process-schema.spec.ts
```